src/c4/C4_MPI.t.hh

```diff
@@ -353,6 +353,15 @@ template <typename T> DLL_PUBLIC_c4 void global_max(T *x, int n) {
                 communicator);
 }
 
+//---------------------------------------------------------------------------//
+
+template <typename T> DLL_PUBLIC_c4 T prefix_sum(T &node_value) {
+  T prefix_sum = 0;
+  MPI_Scan(&node_value, &prefix_sum, 1, MPI_Traits<T>::element_type(), MPI_SUM,
+           communicator);
+  return prefix_sum;
+}
+
```

Comment:
Are there any DbC statements that can be inserted into this function?  Is there a valid range for node_value?